libs/partners/mongodb/langchain_mongodb/chat_message_histories.py

```diff
@@ -114,7 +118,8 @@ def __init__(
     def messages(self) -> List[BaseMessage]:  # type: ignore
         """Retrieve the messages from MongoDB"""
         try:
-            cursor = self.collection.find({self.session_id_key: self.session_id})
+            skip_count = max(0, self.collection.count_documents({}) - self.max_messages) if self.max_messages != None else 0
```

Comment:
Instead of calling `collection.count_documents` each time, is there a more appropriate method to use? e.g., `limit`? `skip` is skipping the first N messages instead of limiting the results to N messages, right?

I think ideally we expose an argument that limits the number of messages returned and name it `history_size` (since as @eyurtsev points out this name has been chosen for similar purposes elsewhere).

Failing that, my preference would be to expose `skip` directly and just let users access it.

Reply:
The query is giving top few results from the starting of conversation If we use `limit`. Instead we would like to have latest messages to append into history.

If we expose `skip`, then user will have to query mongodb database (`collection.count_documents`) every time to count the messages in collection, which is suboptimal solution from point of end user.

I think `history_size` as exposed parameter with default value of 6 or 8 is an ideal condition.

Please let me know your thoughts. I will change the code accordingly and commit.